systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -639,6 +642,141 @@ void testKafkaRebalanceAutoApprovalMechanism() {
+            @Step(value = "Check that topic replicas are moved to the new brokers.", expected = "Topic replicas are distributed onto the newly added brokers."),
+            @Step(value = "Scale Kafka down to the original number of brokers.", expected = "Kafka brokers are scaled down, and Cruise Control initiates rebalancing in REMOVE_BROKERS mode."),
+            @Step(value = "Verify that Kafka auto-rebalance status transitions to RebalanceOnScaleDown and then back to Idle.", expected = "Auto-rebalance status moves to RebalanceOnScaleDown during scaling down and returns to Idle after rebalancing completes."),
+            @Step(value = "Confirm that the cluster is stable after scaling operations.", expected = "Cluster returns to a stable state with initial number of brokers and Cruise Control completed the rebalancing.")
+        },
+        labels = {
+            @Label(value = TestDocsLabels.CRUISE_CONTROL),
+        }
+    )
+    void testAutoKafkaRebalanceScaleUpScaleDown() {
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+        final String scaleUpKafkaRebalanceTemplateName = testStorage.getClusterName() + "-scale-up";
+        final String scaleDownKafkaRebalanceTemplateName = testStorage.getClusterName() + "-scale-down";
+        final int initialReplicas = 3;
+
+        resourceManager.createResourceWithWait(
+            NodePoolsConverter.convertNodePoolsIfNeeded(
+                KafkaNodePoolTemplates.brokerPool(testStorage.getNamespaceName(), testStorage.getBrokerPoolName(), testStorage.getClusterName(), initialReplicas).build(),
+                KafkaNodePoolTemplates.controllerPool(testStorage.getNamespaceName(), testStorage.getControllerPoolName(), testStorage.getClusterName(), initialReplicas).build()
+            )
+        );
+
+        // create two KafkaRebalance with marking resource as configuration template
+        resourceManager.createResourceWithoutWait(
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getNamespaceName(), testStorage.getClusterName())
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_REBALANCE, "template")
+                    .withName(scaleUpKafkaRebalanceTemplateName)
+                .endMetadata()
+                .build(),
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getNamespaceName(), testStorage.getClusterName())
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_REBALANCE, "template")
+                    .withName(scaleDownKafkaRebalanceTemplateName)
+                .endMetadata()
+                .build()
+        );
+
+        // define CC with 2 KafkaRebalance templates
```

Comment:
this is more "define CC with two auto-rebalance modes (add/remove) and corresponding templates"